gms/gossiper.cc

```diff
@@ -775,6 +730,84 @@ future<gossiper::endpoint_permit> gossiper::lock_endpoint(inet_address ep) {
     });
 }
 
+
+future<> gossiper::failure_detector_thread_for_node(gms::inet_address node) {
```

Comment:
This is not a thread. That's good, but it shouldn't claim to be a thread.

Reply:
(it's pretty good that coroutines are cheap enough so we can have one per node without feeling bad; it simplifies the code a lot)

Reply:
Yes, technically it is not a thread, but what it does is really like a thread. We let it run and run, wait for it to finish. What do you suggest to call it,  failure_detector_loop_for_node and failure_detector_loop?

Reply:
Yes, coroutines are awesome here.